sympy/solvers/solvers.py

```diff
@@ -1018,11 +1018,9 @@ def _sympified_list(w):
             ok = True
         else:
             if fi.is_number:
-                if fi.is_Number:
-                    if fi.is_zero:
-                        continue
+                fi = simplify(fi)
+                if not fi.is_zero:
```

Comment:
This is what I propose. Here even if `simplify` is computationally expensive, doing it this way will save us computation by handling all the numerical cases in this block of code and not allowing them to propogate further (and convert into `Poly`) .

Since when expression evaluates to `zero` it'll have no effect on the system as a whole so it can be omitted. If it is non-zero even after a `simplify` call we'll simply return it as an empty list.

Reply:
Sometimes `simplify` can be extremely slow and might even hang. It is possibly better just to give the benefit of the doubt if we are not sure that the expression is zero or not and assume that it is.

Reply:
Alright I'll remove that but on its own `.is_zero` is not able to handle this not sure condition. ( As in the case of sin(2)**2 + cos(2)**2 - 1 which I presume internally `is_zero` evaluates numerically and is unable to classify it as zero ).

Should I instead do a `.evalf(chop = True)` or let this be.

Reply:
I think just let it be. If the equation is not known to be zero or not we can allow it to progress to the next stages of solving.